store/src/java/com/zimbra/cs/dav/resource/MailItemResource.java

```diff
@@ -512,19 +513,18 @@ public void patchProperties(DavContext ctxt, java.util.Collection<Element> set,
         Mailbox mbox = null;
         try {
             mbox = getMailbox(ctxt);
-            mbox.lock.lock();
-            Metadata data = mbox.getConfig(ctxt.getOperationContext(), CONFIG_KEY);
-            if (data == null) {
-                data = new Metadata();
+            try (final MailboxLock l = mbox.lock(true)) {
```

Comment:
You have changed the lifecycle of the lock. Needs to be fixed.

You can see from the old code that the lock is not released until after the `finally` block of the parent `try/catch`